lib/expectable.rb

```diff
@@ -3,30 +3,29 @@
 
 module Expectable
   def expect(method_name, return_value = nil, expected_args = nil)
-    mock = expect_helper(return_value, expected_args)
+    mock = generate_mock(return_value, expected_args)
 
     stub(method_name, mock) do
       yield
     end
 
     begin
       mock.verify
-    rescue
+    rescue MockExpectationError
       raise ExpectationError, "Expected method #{method_name} to be called on object " \
                               "<#{self.class} ##{object_id}>"
     end
   end
 
   private
 
-  def expect_helper(return_value = nil, expected_args = nil)
+  def generate_mock(return_value = nil, expected_args = nil)
     mock = Minitest::Mock.new
 
     # Expected arguments can either be an array or a Proc
-    if expected_args && expected_args.is_a?(Array)
+    if expected_args&.is_a?(Array)
       mock.expect :call, return_value, expected_args
-      # Note: In Ruby lambdas have class Proc
-    elsif expected_args && expected_args.class == Proc
+    elsif expected_args.class == Proc # Note: In Ruby lambdas have class Proc
```

Comment:
If `expected_args` is nil then `expected_args.class` will be `NilClass` and the condition will be false anyway.